src/pages/auth/SignupPage.tsx

```diff
@@ -0,0 +1,170 @@
+import { useState, ChangeEvent, FocusEvent, useEffect } from "react";
+import {
+  AuthContainer,
+  LogoHomeLink,
+  Form,
+  AuthSwitch,
+  SubmitButton,
+} from "./AuthStyles";
+import InputItem from "../../components/UI/InputItem";
+import logo from "../../assets/images/logo/logo.svg";
+import { Link } from "react-router-dom";
+import SocialLogin from "./components/SocialLogin";
+import { SignupInputId, getErrorMessage } from "./authUtils";
+import PasswordInput from "./components/PasswordInput";
+import useDebounce from "../../hooks/useDebounce";
+
+interface FormState {
+  email: string;
+  nickname: string;
+  password: string;
+  passwordConfirmation: string;
+}
+
+interface ErrorState {
+  email?: string;
+  nickname?: string;
+  password?: string;
+  passwordConfirmation?: string;
+}
+
+const SignupPage: React.FC = () => {
+  // 상태 관리를 위해 하나의 object로 묶어주었어요.
+  const [formState, setFormState] = useState<FormState>({
+    email: "",
+    nickname: "",
+    password: "",
+    passwordConfirmation: "",
+  });
+  const [errors, setErrors] = useState<ErrorState>({});
```

Comment:
P1:
FormState와 대응하는 error 값으므로 따로 type을 정의하기보다 아래처럼 사용하시는것이
의도를 더 잘 나타내고 유지보수하기도 좋습니다~

```suggestion
const SignupPage: React.FC = () => {
  // 상태 관리를 위해 하나의 object로 묶어주었어요.
  const [formState, setFormState] = useState<FormState>({
    email: "",
    nickname: "",
    password: "",
    passwordConfirmation: "",
  });
  const [errors, setErrors] = useState<Partial<FormState>>({});
```

https://www.typescriptlang.org/docs/handbook/utility-types.html#partialtype